### ISSUE TITLE
JS | Optional chaining left hand assignment not detected as syntax error.

### ISSUE THREAD

**Issue by joshuao3** [microsoft/TypeScript#57640]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.86.2
- OS Version: Windows 11

Steps to Reproduce:

1. Add this code to a .js file:
```
let tree = {type: 'none'};
tree?.type = 'apple';
```

2. This should be detected as a syntax error. Specifically "Uncaught SyntaxError: Invalid left-hand side in assignment". It is not.

**Comment by uniboxx** [comment]
Hi, I have the 1.87 version and i keep getting the same error.  The worst is that code insert ? automatically so I have to remove it manually every time I operate on objects

**Comment by jakebailey** [comment]
Are you sure it's not being picked up a syntax error? https://www.typescriptlang.org/play?filetype=js#code/DYUwLgBGBOIhBeCBvMBPADiAXBA5AHYD2BIeAvgNwCwAUDHAPwB06Wi+AhhhqHpUA

**Comment by uniboxx** [comment]
Hi looking on internet I read that the chaining operator is not allowed in the left side of assignements. but my problem is still emmet keep inserting ? every time i tab object.method

**Comment by uniboxx** [comment]
For now I solved creating the snippets for .textContent and .innerHTML that I use to make assignements.

**Comment by joshuao3** [comment]
> Are you sure it's not being picked up a syntax error? https://www.typescriptlang.org/play?filetype=js#code/DYUwLgBGBOIhBeCBvMBPADiAXBA5AHYD2BIeAvgNwCwAUDHAPwB06Wi+AhhhqHpUA

Yes. See attached screenshot.

![Screenshot 2024-03-11 090630](https://github.com/microsoft/TypeScript/assets/24870625/03338fdd-7d9d-4a51-9aad-9034c93cc0f8)

**Comment by uniboxx** [comment]
Hi, i found the problem in settings.json, and exactely this line:

"js/ts.implicitProjectConfig.checkJs": true,

putting false or deleting the line the problem disappears

**Comment by Andarist** [comment]
@jakebailey @RyanCavanaugh I don't think this one is actionable. As mentioned above, the problem is reported when the file is checked. While this is a syntax error, TS raises some of them as grammar errors (sorry if my lingo here isn't precise-enough 😉 ) and those are reported by the checker. From the IDE's PoV etc it's nice this code stays parseable in TS. This is no different from **a lot** of other errors like this one:
```ts
// @checkJs: false
// @filename: index.js

class A extends {} // "missing" error: 'extends' list cannot be empty.(1097)
```

**Comment by RyanCavanaugh** [comment]
I think it'd still be preferable if we saw grammar errors in JS, it just means this is a bigger work item than this specific issue
